src/Entities/Minecart.cpp

```diff
@@ -845,24 +845,16 @@ bool cMinecart::TestEntityCollision(NIBBLETYPE a_RailMeta)
 		{
 			if (MinecartCollisionCallback.GetCollidedEntityPosition().z >= GetPosZ())
 			{
-				if ((-GetSpeedZ() * 0.4) < 0.01)
+				if (GetSpeedZ() > 0)  // Check if entity is in front of minecart
```

Comment:
The condition here didn't change (much). GetSpeedZ can be positive or negative depending on the direction the minecart is moving to. We already know in which direction the entity is due to the outer if-clause, so we now need to check if the minecart is moving into that direction.